### PR TITLE
Remotemmi lights

### PR DESCRIPTION
**Pull request by theisjabra**

### CODE REVIEW

File: nodesdk/src/main/device.ts [hunk 1]
```diff
@@ -1206,6 +1207,16 @@ export class DeviceType implements DeviceInfo, DeviceTiming, MetaApi {
         });
     }
```
Comment (extmchristensen) [inline nodesdk/src/main/device.ts:1209]: Need doc comments

Comment (theisjabra) [reply nodesdk/src/main/device.ts:1209]: doc comments added

File: nodesdk/src/main/sdkintegration.d.ts [hunk 2]
```diff
@@ -211,4 +212,7 @@ export declare interface SdkIntegration {
 
     GetButtonFocus(deviceId: number, btnEvents: Array<{ buttonTypeKey: number, buttonTypeValue: string, buttonEventType: Array<{ key: number, value: string }> }>, callback: (error: JabraError, result: void) => void): void;
     ReleaseButtonFocus(deviceId: number, btnEvents: Array<{ buttonTypeKey: number, buttonTypeValue: string, buttonEventType: Array<{ key: number, value: string }> }>, callback: (error: JabraError, result: void) => void): void;
+
+    GetRemoteMMIFocus(deviceId: number, type: enumRemoteMmiType, input: enumRemoteMmiInput, priority: enumRemoteMmiPriority, callback: (error: JabraError, result: void) => void): void;    
+    ReleaseRemoteMmiFocus(deviceId: number, type: enumRemoteMmiType, callback: (error: JabraError, result: void) => void): void;
```
Comment (theisjabra) [inline nodesdk/src/main/sdkintegration.d.ts:217]: I need to change the naming of MMI to be consitent uppercase or PascalCase

**Comment by dlaezza-jabra** [comment]
Thumbs up from my side 👍

File: nodesdk/src/main/device.cc [hunk 3]
```diff
@@ -851,3 +851,122 @@ Napi::Value napi_GetEqualizerParameters(const Napi::CallbackInfo& info) {
...
+  bool argsOk = util::verifyArguments(functionName, info, {util::NUMBER, util::NUMBER, util::OBJECT, util::FUNCTION});
+
+  if (argsOk) {
+    const unsigned short deviceId = (unsigned short)(info[0].As<Napi::Number>().Int32Value());
+    const RemoteMmiType type = (RemoteMmiType)(info[1].As<Napi::Number>().Int32Value());  
+    Napi::Object actionOutputArgs = info[2].As<Napi::Object>();
+    Napi::Function javascriptResultCallback = info[3].As<Napi::Function>();  
+
+    RemoteMmiActionOutput actionOutput;
+    actionOutput.red = util::getObjInt32OrDefault(actionOutputArgs, "red", 0);
```
Comment (theisjabra) [inline nodesdk/src/main/device.cc:953]: RemoteMmiActionOutput struct type is `uint8_t` but [Napi doesn't seem to have a converter for that type](https://github.com/nodejs/node-addon-api/blob/master/doc/number.md). Functionality- and compiler-wise int32_t seems to be ok, but I wonder if I introduce unobvious issues this way?

Comment (extmchristensen) [reply nodesdk/src/main/device.cc:953]: Should be ok.

File: nodesdk/src/main/device.cc [hunk 4]
```diff
@@ -851,3 +851,122 @@ Napi::Value napi_GetEqualizerParameters(const Napi::CallbackInfo& info) {
...
+    actionOutput.green = util::getObjInt32OrDefault(actionOutputArgs, "green", 0);
+    actionOutput.blue = util::getObjInt32OrDefault(actionOutputArgs, "blue", 0);
+    actionOutput.sequence = RemoteMmiSequence::MMI_LED_SEQUENCE_SLOW;
```
Comment (theisjabra) [inline nodesdk/src/main/device.cc:956]: forgot this...

Comment (extmchristensen) [reply nodesdk/src/main/device.cc:956]: also looks good with the int32's but why is sequence hardcoded ?

Comment (theisjabra) [reply nodesdk/src/main/device.cc:956]: will fix

File: nodesdk/src/main/device.ts [hunk 5]
```diff
@@ -45,12 +45,13 @@ export namespace DeviceTypeCallbacks {
     export type onGNPBtnEvent = (btnEvents: Array<{ buttonTypeKey: number, buttonTypeValue: string, buttonEventType: Array<{ key: number, value: string }> }>) => void;
     export type onDevLogEvent = (data: DevLogData) => void;
     export type onBatteryStatusUpdate = (levelInPercent: number, isCharging: boolean, isBatteryLow: boolean) => void;
+    export type onRemoteMmiEvent = (type: number, input: number) => void;
```
Comment (extmchristensen) [inline nodesdk/src/main/device.ts:48]: type and input args should properly be enums

File: nodesdk/src/main/sdkintegration.d.ts [hunk 6]
```diff
@@ -37,6 +37,7 @@ export declare interface SdkIntegration {
                buttonInDataTranslated: (deviceID: number, translatedInData: enumDeviceBtnType, buttonInData: bool) => void,
                devLogCallback: (deviceId: number, json: string) => void,
                batteryStatusCallback: (deviceId: number, levelInPercent: number, isCharging: boolean, isBatteryLow: boolean) => void,
+               onRemoteMmiEvent: (deviceId: number, type: number, input: number) => void,
```
Comment (extmchristensen) [inline nodesdk/src/main/sdkintegration.d.ts:40]: see comment above for type, input args.

**Comment by extmchristensen** [comment]
Overall, looks very good. Just 2 tiny comments. Excellent work. If it works in your own tests on PC and Mac, I think you can merge it into master.

File: nodesdk/src/main/device.ts (hunk 5 above)
Comment (theisjabra) [reply nodesdk/src/main/device.ts:48]: yes good point

**Comment by theisjabra** [comment]
Thanks! Tested on Windows and Mac, and build on Linux - all is ok.

**Pull request merged**